### PR TITLE
Show current and visited nodes

### PR DESCRIPTION
**Pull request by songofroland**

### CODE REVIEW

File: src/logic/graphFrames.ts [hunk 1]
```diff
@@ -7,70 +7,98 @@ import {
...
-    frames.push(createFrameForNode(frames));
-    frames.push(...createFramesForLookups(travelsal, frames));
+  let frames: Array<FrameWithEdgesAndNodes> = [];
+  const visitedNodes : Array<number> = [];
+  for (let travelsal of algorithm.traversalHistory) {
+    const framesWithEdgeData = [
+      createFrameForNode(frames), ...createFramesForLookups(travelsal, frames),
+    ];
+    frames.push(...addNodeFrameData(framesWithEdgeData, travelsal.node, visitedNodes));
+    visitedNodes.push(travelsal.node);
```
Comment (wilmhit) [inline src/logic/graphFrames.ts:35]: visited nodes are kept in frame, why there is separate data structure to keep this?

Comment (wilmhit) [reply src/logic/graphFrames.ts:35]: Just pass in visited nodes from last frame to `addNodeFrameData`

File: src/logic/graphFrames.ts [hunk 2]
```diff
@@ -7,70 +7,98 @@ import {
...
 }
 
+function addNodeFrameData(
+  frames: Array<FrameWithEdges>, 
+  activeNode: number,
+  visitedNodes: Array<number>,
+): Array<FrameWithEdgesAndNodes> 
+{
+  const framesWithNodeData = [];
+  for (let frame of frames){
```
Comment (wilmhit) [inline src/logic/graphFrames.ts:78]: Array.map here would be perfect

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/components/Description.tsx [hunk 3]
```diff
@@ -3,10 +3,11 @@ import React from 'react';
...
     <div className="description">
-      Lorem ipsum dolor sit amet consectetur adipisicing elit.<br/>
-      Quam eveniet placeat mollitia unde iusto labore molestiae,<br/>
-      blanditiis impedit aperiam veniam non, aliquid assumenda,<br/>
-      eaque libero eius ab architecto! Ipsum, veniam.<br/>
+        Dijkstra&apos;s algorithm to find the shortest path between a and b.
+        It picks the unvisited vertex with the lowest distance,<br/>calculates 
+        the distance through it to each unvisited neighbor, and updates the 
+        neighbor&apos;s distance if smaller.<br/> 
+        Mark visited (set to blue) when done with neighbors.
```
[the remark this answers is not in the document]
Comment (songofroland) [reply src/components/Description.tsx:10]: Agree, I've literally pasted the first result from google

File: src/logic/graphFrames.ts (hunk 2 above)
Comment (songofroland) [reply src/logic/graphFrames.ts:78]: Fair point

File: src/logic/graphFrames.ts (hunk 1 above)
Comment (songofroland) [reply src/logic/graphFrames.ts:35]: could u show me what u mean? I don't see it

Comment (wilmhit) [reply src/logic/graphFrames.ts:35]: You use array called `visitedNodes` here. This is unnecessary. You could get this information from `lastFrame.visitedNodes`. It's just a matter of selecting lastFrame. So line 34 could look something like this:
`frames.push(...addNodeFrameData(framesWithEdgeData, travelsal.node, lastFrame.visitedNodes));`

Comment (songofroland) [reply src/logic/graphFrames.ts:35]: You're right, but implementing this we would output a real sculpture as a result.  

Look at line 35, I need to append the current node after each traversal. Also, initially, the frames are empty. Indeed we add an additional data structure but I think this is the most straightforward way to do it. 

If u insist to change it, please paste here a working snippet and if it's more articulate than my solution I'll use it. (I've tried to go with ur idea but it was not clearer)

**Review by ghost — APPROVED** [review]

**Pull request merged**